sources/PanelButton.cpp

```diff
@@ -68,6 +73,14 @@ PanelButton::CreatePanel(int8 which, const BView* target, uint32 flavors,
 		panel = new BFilePanel(B_OPEN_PANEL, &msgr, NULL, flavors, false, NULL,
 			filter);
 
-		panel->Window()->SetTitle(pairs[which].locale);
+		BString title = B_TRANSLATE("Filer");
+		title += ": ";
+
+		if (prefix != NULL)
+			title << prefix << ' ';
```

Comment:
I could have sworn I had commented on line 206 of TextView.cpp below, too...

Whereever you concatenate strings with += or << you have a fixed order of string components. This order can be different in other languages, or they have some weird grammatic rules that demand different word-endings when certain words are combined. All kinds of weird stuff...
Also, if you hand the string snippets to the translators, like B_TRANSLATE("Choose"), B_TRANSLATE("for") etc. it's difficult to find the right words because it's lacking context. It's always better to do something like

```
if (fType == TEST_LOCATION) {
	BString title(B_TRANSLATE_COMMENT("Filer: Choose folder for %prefix%",
		"Don't translate %prefix%, it's the test type 'Type/Name/Size/Location'");
} else {
	BString title(B_TRANSLATE_COMMENT("Filer: Choose file for %prefix%",
		"Don't translate %prefix%, it's the test type 'Type/Name/Size/Location'");
}
title.ReplaceFirst("%prefix%", prefix);
```